app/src/main/java/com/afwsamples/testdpc/common/keyvaluepair/KeyValueBundleArrayFragment.java

```diff
@@ -40,43 +41,50 @@
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment
         .ARG_INITIAL_VALUE;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.ARG_KEY;
+import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment
+        .ARG_RESTRICTION_ENTRY;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.DialogType;
+import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.RESULT_ENTRY;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.RESULT_KEY;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.RESULT_TYPE;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.RESULT_VALUE;
 
 public class KeyValueBundleArrayFragment extends ManageAppFragment implements
-        OnEditButtonClickListener<Bundle>, OnDeleteButtonClickListener<Bundle> {
+        OnEditButtonClickListener<Integer>, OnDeleteButtonClickListener<Integer> {
     /**
      * Key of the editing bundle array.
      */
     private String mKey;
     /**
      * The current bundle list.
      */
-    private List<Bundle> mBundleList;
+    private ArrayList<Object> mBundleList;
```

Comment:
Thanks for review.
Regarding ArrayList - yes, it can be changed to List.
Regarding Bundle and RestrictionEntry, looks like we really had some misunderstanding. In your old comment I see following: "How about having the fragment takes both value(Bundle) and format(RestrictionEntry) as input and render accordingly? **With this, we can remove all the conversion code too.**"
 But we cannot remove conversion code, if we will pass both Bundle and RestrictionEntry list as parameters for app restrictions. We will need to convert manifest restrictions into bundle, at least when application has no any restrictions values saved in DevicePolicyManager and manifest restrictions only are available.
 Also, passing same data twice (as Bundle list and as RestrictionEnty list) looks strange.
 One more point - with using Bundle list it will be difficult to show restrictions in same order as defined by application. But this is important, because in case of complex restrictions, application developer can define them in some specific order. For example, if restrictions defines some account settings, then user name probably will be first restriction to be dsplayed to user. But in case of alphabetic sorting it will be near the end of the restrictions list.
  Yes, there is some "if - else" code to handle both Bundle and RestrictionEntry, but it was added to support Bundles for SetTrustAgentConfigFragment only. For application restrictions it is much more convenient to use RestrictionEntry list and convert it into Bundle for saving into DevicePolicyManager only.